### DISCUSSION TITLE
Add support for SPI pictures for Ensemble Label

### DISCUSSION THREAD

**Discussion by andimik** [KejPi/AbracaDABra#168] [Ideas]
I've realized that Croatian broadcaster OiV also includes their own logos into the SPI directory, whereas all (?) others in Europe only use it for the radio program logos and EPG.

![eiE3CA030](https://github.com/user-attachments/assets/3017a870-207f-4368-8ea6-f99108d3ce56)
![eiE3CA031](https://github.com/user-attachments/assets/f6b6d0ff-00e3-40d8-90c0-2bc7054e7d47)
![eiE3CA032](https://github.com/user-attachments/assets/8464a77a-ca9d-488e-ab6b-78be8bedaafe)
![eiE3CA033](https://github.com/user-attachments/assets/69e532cf-ec18-4474-9964-347530c7e496)

[e3ca03.zip](https://github.com/user-attachments/files/17443867/e3ca03.zip)

```
  <shortName><![CDATA[OIV DAB+]]></shortName>
  <mediumName><![CDATA[OIV Croatia DAB+]]></mediumName>
  <mediaDescription>
   <multimedia url="eiE3CA031.png" mimeValue="image/png" height="128" width="128" type="logo_unrestricted"/>
  </mediaDescription>
  <mediaDescription>
   <multimedia url="eiE3CA030.png" mimeValue="image/png" height="240" width="320" type="logo_unrestricted"/>
  </mediaDescription>
  <mediaDescription>
   <multimedia url="eiE3CA032.png" type="logo_colour_square"/>
  </mediaDescription>
  <mediaDescription>
   <multimedia url="eiE3CA033.png" type="logo_colour_rectangle"/>
  </mediaDescription>
```

As this is not an issue (but an idea) I wonder where we could integrate it without breaking the layout.

[2024-06-30_120851_11D.uff.zip](https://github.com/user-attachments/files/17444077/2024-06-30_120851_11D.uff.zip)
(longer file truncated to 20 MB, but should be enough)

Hint: I'm aware that there is no small logo for HR1 and HR2, see xml file.

**Comment by KejPi** [comment]
I know about that feature in SPI, but I think it is rarely used. Technically it is not a big problem to support it, the real problem is how to visualize it in current application layout as you have mentioned.

**Comment by andimik** [comment]
Well, you could display it in the mouse over, like you also do for service names.

**Comment by KejPi** [comment]
I have checked the recordings I have and none contains ensemble logo. It seems to be very rare usecase. Do you know about any other ensemble with the logo in SPI?

**Comment by andimik** [comment]
Well, I also found another ensemble beside of Croatian's OiV, it's Norwegian NRK (Regions 1 till 7).

```
<shortName><![CDATA[NRK Reg1]]></shortName>
  <mediumName><![CDATA[NRK Reg1 OsloVik]]></mediumName>
  <mediaDescription>
   <multimedia height="240" mimeValue="image/png" type="logo_unrestricted" url="eiE2F7010.png" width="320"/>
   <multimedia height="128" mimeValue="image/png" type="logo_unrestricted" url="eiE2F7011.png" width="128"/>
   <multimedia type="logo_colour_square" url="eiE2F7012.png"/>
   <multimedia type="logo_colour_rectangle" url="eiE2F7013.png"/>
  </mediaDescription>
```

[4079_48579.zip](https://github.com/user-attachments/files/19812206/4079_48579.zip)

![eiE2F7010](https://github.com/user-attachments/assets/2bf5d710-4254-4030-8e06-ddee158b6496)
![eiE2F7011](https://github.com/user-attachments/assets/296618ed-29cb-48fa-b0dd-b5e180b3382b)
![eiE2F7012](https://github.com/user-attachments/assets/7568f63a-301c-4cc6-be4a-b4910353c3b4)
![eiE2F7013](https://github.com/user-attachments/assets/fc720b56-7a19-4440-9a24-9c1f2e292673)

This is a zipped raw file (6 seconds, but should be enough using the SPI data from above zip-file)

[nrk1_20221227.raw.zip](https://github.com/user-attachments/files/19812275/nrk1_20221227.raw.zip)

**Reply by andimik** [reply]
And one local from Germany

```
  <shortName><![CDATA[DAB+ BKH]]></shortName>
  <mediumName><![CDATA[DAB+ Mux Bad KH]]></mediumName>
  <mediaDescription>
   <multimedia height="240" mimeValue="image/png" type="logo_unrestricted" url="E0_1007_0_320_240.png" width="320"/>
   <multimedia height="128" mimeValue="image/png" type="logo_unrestricted" url="E0_1007_0_128_128.png" width="128"/>
   <multimedia type="logo_colour_rectangle" url="E0_1007_0_112_32.png"/>
   <multimedia type="logo_colour_square" url="E0_1007_0_32_32.png"/>
  </mediaDescription>
```

![E0_1007_0_128_128](https://github.com/user-attachments/assets/fc39d6d9-5bb3-4ba5-9e74-627f629cd898)
![E0_1007_0_320_240](https://github.com/user-attachments/assets/e710169a-420b-40b0-bb1b-401fa401ab51)
![E0_1007_0_112_32](https://github.com/user-attachments/assets/20e6202c-7808-448e-af18-072b148edce2)
![E0_1007_0_32_32](https://github.com/user-attachments/assets/09c3db84-1674-4af7-a0b6-a42d2ac66b9d)

**Reply by KejPi** [reply]
Well, it is not enough for implementation. I would need some IQ recording with SPI directory that can be decoded (minimum is XML and ensemble logo files that can be decoded from the recording).

**Reply by andimik** [reply]
ok, here is one from Sweden, this is a **raw** file (174 seconds)

 https://we.tl/t-PjR4DFuzIe 

![grafik](https://github.com/user-attachments/assets/03b55898-08ee-4c5a-a23e-045a0c5f9803)


![eiE3E1050](https://github.com/user-attachments/assets/eb6d0cc7-89b4-42f9-91e7-e2008d6e52f7)
![eiE3E1051](https://github.com/user-attachments/assets/97e6edf3-6d15-4a0f-b771-bb4e09aefd57)
![eiE3E1052](https://github.com/user-attachments/assets/7652f21f-87bf-43dd-b460-7e26acd922cb)
![eiE3E1053](https://github.com/user-attachments/assets/248a8601-84f1-454c-986e-4ba2a8355182)

```
<shortName><![CDATA[SR MALMÖ]]></shortName>
  <mediumName><![CDATA[SR MALMÖ]]></mediumName>
  <mediaDescription>
   <multimedia height="128" mimeValue="image/png" type="logo_unrestricted" url="eiE3E1051.png" width="128"/>
  </mediaDescription>
  <mediaDescription>
   <multimedia height="240" mimeValue="image/png" type="logo_unrestricted" url="eiE3E1050.png" width="320"/>
  </mediaDescription>
  <mediaDescription>
   <multimedia type="logo_colour_square" url="eiE3E1052.png"/>
  </mediaDescription>
  <mediaDescription>
   <multimedia type="logo_colour_rectangle" url="eiE3E1053.png"/>
  </mediaDescription>
```

**Reply by KejPi** [reply]
That's perfect, thank you. I am going to add this feature in next version.

**Reply by andimik** [reply]
this is the obove mentioned file from Norway, the size is 1.2 GB (zipped 687 MB) and the length is 283 seconds

see this zipped **raw file**
https://www.transfernow.net/dl/20250418x5iNwjhj

(I had to look for an alternative for wetransfer, as the file upload was too slow and started with 0 again and again)

**Reply by KejPi** [reply]
Great, thanks! I this should be enough for the implementation :-)

**Reply by andimik** [reply]
I know, but both are quite unique and could be used also in the future: 

- very high bitrate in Sweden (more than 144 CUs)
- real EPG with lots of days and services
- the funny thing is, the Swedish raw file is fixed to 12B (never seen it before). All other raw files can be loaded from 5A to 13F. Is this an undocumented feature? It's NOT an xml file, so there is NO header

**Reply by KejPi** [reply]
Yes, it is a feature - application checks for channel in filename and if it is found (12B in this case) then it uses it for tuning.

**Reply by andimik** [reply]
Oh God, thank you for the explanation. I didn't realize that the file name is the trigger... 😁

**Comment by KejPi** [comment]
So, what about this? 

<img width="924" alt="Snímek obrazovky 2025-04-18 v 17 57 15" src="https://github.com/user-attachments/assets/512194b9-f1d1-45f9-90fd-725248d35b29" />

**Reply by andimik** [reply]
I like that!

**Comment by andimik** [comment]
Remark: By chance, I found out that also the new Irish mux Fáilte supports it, **just by checking the digitalbitrate website** ;-) .

https://www.digitalbitrate.com/dtv.php?mux=5B&cpid=2700869631&live=640&sec=0&lang=en

![grafik](https://github.com/user-attachments/assets/d5be6f4f-523c-4a85-9d47-e9c8b65c26b6)
![grafik](https://github.com/user-attachments/assets/66454ae7-83d2-47a6-857d-dfcd2f47c04b)

**Comment by KejPi** [comment]
Implemented in 3.2.0
